src/main/java/shop/service/impl/CsvWriterImpl.java

```diff
@@ -0,0 +1,27 @@
+package shop.service.impl;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import shop.service.ReportService;
+import shop.service.Writer;
+
+public class CsvWriterImpl implements Writer {
+    private final ReportService reportService;
+
+    public CsvWriterImpl() {
+        reportService = new ReportServiceImpl();
+    }
```

Comment:
you should pass the report to `write()` method
and call the method of ReportService in main()

because your service should not depend on one another